pkg/operator/volumesnapshotclasscontroller/volumesnapshotclasscontroller.go

```diff
@@ -0,0 +1,88 @@
+package volumesnapshotclasscontroller
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/openshift/alibaba-disk-csi-driver-operator/pkg/alibaba"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	configinformer "github.com/openshift/client-go/config/informers/externalversions/config/v1"
+	infralisterv1 "github.com/openshift/client-go/config/listers/config/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceread"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/klog/v2"
+)
+
+type VolumeSnapshotClassController struct {
+	manifest       []byte
+	infraLister    infralisterv1.InfrastructureLister
+	dynamicClient  dynamic.Interface
+	operatorClient v1helpers.OperatorClient
+	recorder       events.Recorder
+}
+
+func NewVolumeSnapshotClassController(
+	name string,
+	manifest []byte,
+	infraInformer configinformer.InfrastructureInformer,
+	dynamicClient dynamic.Interface,
+	operatorClient v1helpers.OperatorClient,
+	resyncInterval time.Duration,
+	recorder events.Recorder) factory.Controller {
+
+	c := &VolumeSnapshotClassController{
+		manifest:       manifest,
+		infraLister:    infraInformer.Lister(),
+		dynamicClient:  dynamicClient,
+		operatorClient: operatorClient,
+		recorder:       recorder.WithComponentSuffix(name),
+	}
+
+	return factory.New().WithSync(c.sync).ResyncEvery(resyncInterval).WithSyncDegradedOnError(operatorClient).WithInformers(
+		operatorClient.Informer(),
+		infraInformer.Informer(),
```

Comment:
Too bad we won't sync when users change VolumeSnapshotClass objects (we don't have an informer for that, right?)